src/main/java/org/opensearch/security/action/tenancy/DefaultTenantRestHandler.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.tenancy;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.security.privileges.PrivilegesEvaluator;
+
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+public class DefaultTenantRestHandler extends BaseRestHandler {
+
+    private final PrivilegesEvaluator evaluator;
+
+    public DefaultTenantRestHandler(PrivilegesEvaluator evaluator) {
+        super();
+        this.evaluator = evaluator;
+    }
+
+    @Override
+    public String getName() {
+        return "Multi Tenancy actions for Retrieve / Update default tenant.";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+                new Route(GET, "/_plugins/_security/api/tenancy/default_tenant"),
+                new Route(PUT, "/_plugins/_security/api/tenancy/default_tenant")
+        );
+    }
+
+    public void validate(final RestRequest request) throws Exception {
+
+        boolean isPrivateTenantEnabled = this.evaluator.privateTenantEnabled();
+        boolean isMultitenancyEnabled = this.evaluator.multitenancyEnabled();
+
+        if( !isMultitenancyEnabled ) {
+            throw new IllegalArgumentException("Default tenant can not be set if multi-tenancy is disabled.");
+        }
+
+        JSONParser parser = new JSONParser();
+        JSONObject request_content = (JSONObject) parser.parse(request.content().utf8ToString());
+
+        String default_tenant = (String) request_content.get("value");
+
+        if(default_tenant.equals(""))
+        {
+            return;
+        }
+
+        if(default_tenant.equals("Private")) {
+            if(!isPrivateTenantEnabled)
+            {
+                throw new IllegalArgumentException("Default tenant can not be set to Private if Private tenant is disabled.");
```

Comment:
Isn't the private tenant denoted as `__private__`?

Reply:
There is some ambiguity around that. For eg: [Here](https://github.com/opensearch-project/security-dashboards-plugin/blob/b3b3120144e50d229bb09aec50991a383df6dafb/public/apps/configuration/utils/tenant-utils.tsx#L116)

When (tenant === '__user__') we resolve tenant name to 'Private'. I will be using the same in frontend changes.